ui/app/components/PolicyComponent/PolicyComponent.js

```diff
@@ -40,10 +40,10 @@ class PolicyComponent extends Component {
     const value = parseInt(target.value);
     const id = target.id;
     let schedule = {...this.props.versionedPolicyDetails.data.policyDetails.policy.schedule};
-    if(value){
-      schedule[id] = value;
-    } else{
+    if (isNaN(value)) {
```

Comment:
Why are are we handling a NaN case. Is the error msg shown by the required field not being present in the schedule object?

Reply:
If we don't handle the NaN, it is then handled by the input html tag directly, which throws a warning. The handling which we are doing is to avoid the NaN getting to the input field, instead we should either assign an empty string to the field or if it is an optional field remove the field itself from the payload. Will add this optional-field based conditional logic. 
The only cases where we can get a NaN from parseInt (given that the target values comes from a number type input field) is when user types 'e','E', '-', or '+'.

Reply:
Now using input of type text which allows typing of any characters, but has the regex based validation. This helps us avoiding the parseInt based checks and broken css in cases of inputs including 'e', 'E', ...